idaes/core/util/config.py

```diff
@@ -107,6 +107,33 @@ def list_of_floats(arg):
     return lst
 
 
+def list_of_ints(arg):
+    '''Domain validator for lists of integers
+
+    Args:
+        arg : argument to be cast to list of ints and validated
+
+    Returns:
+        List of ints
+    '''
+    lst = []
+    try:
+        iter(arg)
+    except TypeError:
+        arg = [arg]
+
+    for v in arg:
```

Comment:
my preferred strategy is to try to convert to a int and raise an error if it fails. This way anything that implements `__int__` would be ok.
e.g. 
```
try:
    i = int(v)
except (ValueError, TypeError) as e:
    try:
        i = int(float(v))
    except (ValueError, TypeError) as e:
        raise RuntimeError(f"fuggedaboutit with that {v}")
```

Reply:
I believe I did it this way to explicitly exclude `bools` (which can be cast to `ints`). I am not sure if that is a good idea or not, but the idea was that if you set a `bool` where an `int` was expected it was more than likely that you got something wrong (conceptually at least).

Reply:
I actually prefer the following:
```
i = int(v)
if i != float(v):
    raise ValueError(f"Value {v} is not an integer")
return i
```